core-java-modules/core-java-nio-2/src/main/java/com/baeldung/lock/FileLocks.java

```diff
@@ -0,0 +1,239 @@
+	 * @return
+	 * @throws IOException
+	 */
+	static FileLock getReadLockFromInputStream(long from, long size) throws IOException {
+		Path path = Files.createTempFile("foo", "txt");
+		try (FileInputStream fis = new FileInputStream(path.toFile());
+				FileLock lock = fis.getChannel().lock(from, size, true)) {
+			if (lock.isValid()) {
+				log.debug("This is a valid shared lock");
+				return lock;
+			}
+			return null;
+		}
+	}
+	
+	
+	/**
+	 * Getting an exclusive lock from a RandomAccessFile works if the file is in read mode.
+	 * @param from beginning of the locked region
+	 * @param size how many bytes to lock
+	 * @return
+	 * @throws IOException
+	 */
+	static FileLock getReadLockFromRandomAccessFile(long from, long size) throws IOException {
+		Path path = Files.createTempFile("foo", "txt");
+		try (RandomAccessFile file = new RandomAccessFile(path.toFile(), "r"); // could also be "rw", but "r" is sufficient for reading
+				FileLock lock = file.getChannel().lock(from, size, true)) {
+			if (lock.isValid()) {
+				log.debug("This is a valid shared lock");
+				return lock;
+			}
+			return null;
+		} catch (Exception e) {
+			log.error(e.getMessage());
+		}
+		return null;
+	}
+
+	
+
+	static class Writer implements Runnable {
+
+		private Path path;
+
+		private volatile RandomAccessFile file;
+
+		private String text;
+
+		private volatile CountDownLatch countDownLatch;
+
+		/**
+		 * 
+		 * @param path           The path to the file we will write into
+		 * @param text           The text to write
+		 * @param countDownLatch A counter for thread synchronization
+		 * 
+		 */
+		public Writer(Path path, String text, CountDownLatch countDownLatch) {
+			this.path = path;
+			this.text = text;
+			this.countDownLatch = countDownLatch;
+		}
+
+		@Override
+		public void run() {
+			try {
+				lockAndWrite();
+			} catch (InterruptedException | FileNotFoundException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+			countDownLatch.countDown();
+		}
+
+		private void lockAndWrite() throws InterruptedException, FileNotFoundException {
+			ByteBuffer buffer = null;
+			if (file == null) {
+				file = new RandomAccessFile(path.toFile(), "rw");
+			}
+			try (FileChannel channel = file.getChannel()) {
+
+				try (FileLock lock = channel.tryLock(channel.size(),
+						channel.size() + text.length() + System.lineSeparator().length(), true)) {
+					if (lock != null) {
+						String text = "Hello, world.";
+						buffer = ByteBuffer.allocate(text.length() + System.lineSeparator().length());
+						buffer.put((text + System.lineSeparator()).getBytes(Charsets.UTF_8));
+						buffer.flip();
+						while (buffer.hasRemaining()) {
+							channel.write(buffer, channel.size());
+						}
+					}
+				} catch (OverlappingFileLockException e) {
+					// Failed to lock. Try again later.
+					Thread.sleep(50);
+					lockAndWrite();
+				}
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
```

Comment:
Can we remote the TODO type statements please.

Reply:
I removed the code that wasn't used in unit tests altogether since that was a remainder from previous versions of the article and wasn't being referenced in the article.